mono/metadata/icall.c

```diff
@@ -7157,6 +7157,52 @@ mono_array_get_byte_length (MonoArrayHandle array)
 	}
 }
 
+void
+ves_icall_System_Buffer_BlockCopy (MonoArrayHandle src, int src_offset, MonoArrayHandle dst, int dst_offset, int count, MonoError *error)
+{
+	MONO_CHECK_ARG_NULL_HANDLE (src, FALSE);
+	MONO_CHECK_ARG_NULL_HANDLE (dst, FALSE);
+
+	if (src_offset < 0 || dst_offset < 0 || count < 0) {
+		mono_error_set_index_out_of_range (error);
+		return;
+	}
+
+	MonoClass * const src_class = m_class_get_element_class (MONO_HANDLE_GETVAL (src, obj.vtable)->klass);
+	MonoClass * const dst_class = m_class_get_element_class (MONO_HANDLE_GETVAL (dst, obj.vtable)->klass);
+
+	if (!m_class_is_valuetype (dst_class)) {
+		mono_error_set_argument (error, "dst", "Object must be an array of primitives.");
+	}
+
+	if (!m_class_is_valuetype (src_class)) {
+		mono_error_set_argument (error, "src", "Object must be an array of primitives.");
+	}
+
+	if ((src_offset > mono_array_get_byte_length (src) - count) || (dst_offset > mono_array_get_byte_length (dst) - count)) {
+		mono_error_set_argument (error, "", "Offset and length were out of bounds for the array or count is greater than the number of elements from index to the end of the source collection.");
+		return;
+	}
+
+	MONO_ENTER_NO_SAFEPOINTS;
+
+	const guint8* src_buf =  (guint8*)MONO_HANDLE_RAW (src)->vector + src_offset;
+	const guint8* dest_buf = (guint8*)MONO_HANDLE_RAW (dst)->vector + dst_offset;
+
+	memmove (dest_buf, src_buf, count);
+
+	MONO_EXIT_NO_SAFEPOINTS;
+
+	return;
+}
+
+MonoBoolean
+ves_icall_System_Buffer_IsPrimitiveTypeArray (MonoArrayHandle array, MonoError* error)
+{
+	MonoClass * const klass = m_class_get_element_class (MONO_HANDLE_GETVAL (array, obj.vtable)->klass);
```

Comment:
This should be a JIT intrinsics, will look into it.

Reply:
Ok, I'll remove it once your PR appear